### ISSUE TITLE
Part doesn't delete by TTL

### ISSUE THREAD

**Issue by k-morozov** [ClickHouse/ClickHouse#48144]
There is a part in system.parts that should have been removed in January by TTL.

```
delete_ttl_info_min - 2023-01-13 09:23:37.
```

In system.replication_queue I found:

```
num_postponed:          186842
postpone_reason:        Not executing log entry queue-0003190419 of type MERGE_PARTS for part all_0_40536039_13539_11171721 because source parts size (77.63 GiB) is greater than the current maximum (71.12 GiB).
```

I have default config max_bytes_to_merge_at_max_space_in_pool (150GiB).
I try DROP PART, DETACH PART - It doesn't work.

Maybe someone has another idea how I can remove this large part?

**Comment by k-morozov** [comment]
I don't find rws from first querry and unreservedspace from second querry.

> select from system.parts 
Row 1:
──────
name:                all_0_40536039_13538_11171721
database:            ***
table:               ***
delete_ttl_info_min: 2023-01-13 09:23:37
delete_ttl_info_max: 2023-02-09 14:48:05
bytes_on_disk:       83359738762
`
> select name, formatReadableSize(total_space), formatReadableSize(free_space) from system.disks \G;

`Row 1:
──────
name:                            default
formatReadableSize(total_space): 350.00 GiB
formatReadableSize(free_space):  135.21 GiB

`

**Comment by den-crane** [comment]
>I don't find rws from first querry and unreservedspace from second querry.

They are typos. `rows` & `unreserved_space`

>bytes_on_disk:       83359738762   --> because source parts size (77.63 GiB)
>formatReadableSize(free_space):  135.21 GiB   -> is greater than the current maximum (71.12 GiB)

135.21 / 2 = 67.605 --> new maximum is 67 GB.

Merge needs twice space of total size of parts participated in the merge.
In your case TTL needs  77.63*2 = 155.26 GB free disk space.

-----

>I try DROP PART, DETACH PART - It doesn't work.

As you can see part has data 2023-01-13 - 2023-02-09.
To drop this part execute `alter table ... drop part 'all_0_40536039_13538_11171721'`

**Comment by den-crane** [comment]
>DB::Exception: Part all_0_40536039_13538_11171721 has already been assigned a merge into all_0_40536039_13539_11171721. (PART_IS_TEMPORARILY_LOCKED)

Right. This part is blocked by the planned merge.

The easiest way is to increase the disk space. To make free disk space 160GB.
Check how many space is used by system logs  
https://kb.altinity.com/altinity-kb-useful-queries/altinity-kb-database-size-table-column-size/
https://kb.altinity.com/altinity-kb-setup-and-maintenance/altinity-kb-system-tables-eat-my-disk/

The hard way is to remove planned merges and in the short period of time to drop part until the merge is planned back.

The third way is to stop SELECTs and INSERTs and execute 
```
alter table ... detach partition id 'all'; 
rm -rf detached/all_0_40536039_13538_11171721
alter table ... attach partition id 'all'; 
```

**Comment by den-crane** [comment]
BTW, Is it MergeTree or ReplicatedMergeTree ?
